### DISCUSSION TITLE
Is there a way to get an empty line after entering a function

### DISCUSSION THREAD

**Discussion by lokesh-krishna** [R-nvim/R.nvim#218] [Q&A]
Thanks to completion, whenever I start typing a function name, I can quickly get to `function()` with my cursor position between the parentheses.
However, hitting enter on it results in: 
```
function(
|)
```
With the cursor at the start of the new line and right before the closing parenthesis.

Is there a way to get it to behave as such upon hitting enter from within the parentheses:

```
function(
  |
)
```

Hope I'm making sense 🙈

**Comment by jalvesaq** [comment]
What you are describing is a snippet, not a simple completion. You can distinguish snippets from normal completions because the former ends in `~`.

`R.nvim` completions are provided by [cmp-r](https://github.com/R-nvim/cmp-r) which doesn't have snippets. You are getting snippets from the [languageserver](https://github.com/REditorSupport/languageserver).

**Reply by lokesh-krishna** [reply]
I used `function()` as an example to represent functions. I'm actually talking about the cmp-r completions, such as `mutate()`, `factor()`, etc.

**Comment by jalvesaq** [comment]
The indentation after pressing `<Enter>` is defined by Neovim which has two distinct systems for this: Vim's indent script and tree-sitter-based indentation. Tree-sitter-based indentation is disabled by default. If you don't enable it, your code will be indented by Vim indent scripts.
